Improve Handbook

gzg opened the pull request:
## Related Issues
 Original issue: #3, https://github.com/InteractionDesignFoundation/IDF-web/issues/12071

## Context and Purposes
We wanted to make our handbook more developer-centric and our company more transparent.

## Challenges and Solutions
The biggest challenge was the amount of information that *may* go into the handbook. An example to this is the Benefits. I wanted to create a `benefits` directory explaining company benefits a bit of in detail but then I'd be somehow duplicating that content from the careers page and making it more detailed.

I foresaw that loading a lot of information into the handbook will take a lot of time -- especially also thinking about the review process as well. That's why I decided not to do that.

I focused my attention solely to make the handbook more developer centric and more structured. Here is my changes in general:

1. All documents are put into 3 main directories: `company`, `guides`, and `library`.
2. I added `benefits` under `company` with a link to our careers page.
3. I collected scrum and onboarding documents under their respective directories under `guides`.
4. I moved necessary workflows to under `guides`.
5. I created "Clean Code JS" under frontend library.
6. By having a clear folder structure, I had the chance to simplify document names.
7. I created new Table of Contents across the handbook for easy navigation.
8. I moved "Zen Code" to the handbook's homepage. This actually belonged to `philosophy` and I'm not sure about this change. The idea is to make it immediately accessible and give any reader a sneak peek of what's inside IDF development.
9. More simplifications on the content and links...

## What's next?
We can make the company more transparent one step at a time. So my plan is to not add any more information at the time being. Let's do it in time while the information becomes available & necessary.

## Expected feedback
- [ ] General feedback

lptn commented on guides/onboarding/README.md:
```diff
@@ -0,0 +1,9 @@
+==========
+
+## Table of Contents
+ - [New developer!](developer.md)
```
![image](https://user-images.githubusercontent.com/5278175/46071599-11e99080-c189-11e8-9919-e612bf15667f.png)


I prefer to have `Onboarding` as simple and predictable name

lptn commented on README.md:
```diff
@@ -10,19 +10,17 @@ To achieve this, we want everyone to understand what is expected of them, the th
-    - [Team-trips](/workflows/team-trips/README.md)
+ - [Company](/company/README.md)
+ - [Guides](/guides/README.md)
+ - [Library](/library/README.md)
```
Note that in prev. implementation it was a great starting point for a developer:
![image](https://user-images.githubusercontent.com/5278175/46072142-47db4480-c18a-11e8-9a7b-5bd628f2a981.png)


In this implementation we use only a single hierarchy level and it's not very easy to find an information you need

lptn commented on guides/README.md:
```diff
@@ -0,0 +1,19 @@
+Developer Guides
+================
+
+Welcome aboard!
```
I think it is appropriate for onboarding only

gzg commented:
Hey @lptn 

I addressed your concerns on

- the name "Onboarding" by using the name onboarding and moving the welcome message to "Onboarding" directory,
- the main "Table of Contents" by increasing the nesting level to 1.

Thanks! 🎉 

> I prefer to think about this handbook as about a book that MAY include different roles, but technical ones mostly (e.g. QA).

(1) I agree with you; however we don't currently have more roles and we don't know when to add more roles. I prefer to restructure it at the time we add a new role -- that way, we can make better decisions on how to do it.

> In case if you agree with me -- we should remove zen code from main readme. And move some files

(2) Whenever we add a new role, I assume our Zen Code may adapt to cover that role -- as well as the new role is technical of course. I prefer to have Zen Code as the main mantra around this handbook and update it as the needs change.

What do you think about (1) and (2)? :-)

lptn commented:
(1) (2), ok
